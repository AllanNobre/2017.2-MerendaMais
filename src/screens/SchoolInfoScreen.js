import React from 'react';
import axios from 'axios';
import openMap from 'react-native-open-maps';
import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
import { SideMenu } from 'react-native-elements';
import { Actions } from 'react-native-router-flux';
import Menu from '../components/Menu';
import { SCHOOL_ENDPOINT } from '../constants';
import { logInfo, logWarn } from '../../logConfig/loggers';

const FILE_NAME = 'SchoolInfoScreen.js';

const sideMenuIcon = require('../images/ic_menu_black_48dp_1x.png');
const BackIcon = require('../images/ic_keyboard_arrow_left_48pt.png');
// const Location = require('../images/ic_place.png');

const styles = StyleSheet.create({
  headerBox: {
    flex: 1,
    padding: 10,
    backgroundColor: '#FF9500',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textLogo: {
    flex: 1,
    fontSize: 30,
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  buttonContainer: {
    paddingVertical: 10,
    borderWidth: 1,
    borderRadius: 7,
    marginHorizontal: 15,
    marginTop: 10,
    marginBottom: 10,
    backgroundColor: '#FF9500',
    justifyContent: 'flex-end',
  },

  buttonText: {
    textAlign: 'center',
    color: '#FFF',
  },

  listInfo: {
    flex: 4,
    marginHorizontal: 10,
    paddingTop: 5,
    // justifyContent: 'center',
    width: 340,
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 7,
  },
  text: {
    flexDirection: 'row',
    color: '#95a5a6',
    fontSize: 20,
    paddingTop: 10,
    paddingBottom: 5,
  },

});

class SchoolInfoScreen extends React.Component {
  constructor(props) {
    super(props);

    console.log(this.props);

    this.state = {
      isOpen: false,
      isLoading: false,
      name: '',
      phone: '',
      email: '',
    };
  }

  componentWillMount() {
    logInfo(FILE_NAME, 'componentWillMount',
      `URL:${SCHOOL_ENDPOINT}/${this.props.school.schoolCode}`);

    axios.get(`${SCHOOL_ENDPOINT}/${this.props.school.schoolCode}`, {
      params: {
        campos: 'nome,email,telefone,latitude,longitude',
      },
    })
      .then((response) => {
        logInfo(FILE_NAME, 'componentWillMount',
          `Successfully got school data: ${JSON.stringify(response.data, null, 2)}`);

        // Since we get the response in portuguese, we need to "translate" it so we
        // can add it in the store.
        this.props.setSchoolInfo(
          {
            schoolCode: response.data.codEscola,
            schoolName: response.data.nome,
            schoolPhone: response.data.telefone,
            schoolEmail: response.data.email,
            schoolLat: response.data.latitude,
            schoolLong: response.data.longitude,
          });
      })
      .catch((error) => {
        logWarn(FILE_NAME, 'componentWillMount', error);
      });
  }

  showLocalizationButton() {
    if (this.props.school.schoolLat !== undefined) {
      return (
        <View key="renderButton">
          <Text style={{ color: '#95a5a6', fontSize: 20 }}>Localização: </Text>
          <TouchableOpacity
            onPress={() => this.goToMaps()}
            style={styles.buttonContainer}
            activeOpacity={0.7}
          // <Image source={Location} />
          >
            <Text style={styles.buttonText}>Ver no Mapa</Text>
          </TouchableOpacity>
        </View>
      );
    }

    // If we can't return the button, return nothing.
    return (null);
  }

  updateMenuState(isOpen) {
    this.setState({ isOpen });
  }

  goToMaps() {
    // const url = `https://www.google.com/maps/?q=${this.props.school.schoolLat},${this.props.school.schoolLong}`;
    // Linking.openURL(url);
    openMap({ latitude: this.props.school.schoolLat, longitude: this.props.school.schoolLong });
  }

  render() {
    const menu = <Menu />;
    return (
      <SideMenu
        menu={menu}
        menuPosition="right"
        isOpen={this.state.isOpen}
        disableGestures
        onChange={isOpen => this.updateMenuState(isOpen)}
      >
        <View style={{ flex: 1, backgroundColor: 'white' }}>
          <View style={styles.headerBox}>
            <TouchableOpacity
              onPress={() => Actions.pop()}
            >
              <Image source={BackIcon} />
            </TouchableOpacity>
            <Text style={styles.textLogo}>Pesquisar Escola</Text>

            <TouchableOpacity
              onPress={() => this.updateMenuState(true)}
            >
              <Image source={sideMenuIcon} />
            </TouchableOpacity>
          </View>

          <Text style={styles.text}>  Infomações</Text>
          <View style={styles.listInfo}>
            <Text style={{ color: '#95a5a6', fontSize: 20 }}>Nome: </Text>
            <Text style={{ color: 'black', fontSize: 19 }}>{this.props.school.schoolName}</Text>
            <Text style={{ color: '#95a5a6', fontSize: 20 }}>Email: </Text>
            <Text style={{ color: 'black', fontSize: 19 }}>{this.props.school.schoolEmail}</Text>
            <Text style={{ color: '#95a5a6', fontSize: 20 }}>Telefone: </Text>
            <Text style={{ color: 'black', fontSize: 19 }}>{this.props.school.schoolPhone}</Text>
            {this.showLocalizationButton()}
          </View>

          <TouchableOpacity
            onPress={() => Actions.popTo('schedulingVisit')}
            style={styles.buttonContainer}
            activeOpacity={0.7}
          >
            <Text style={styles.buttonText}>Agendar Visita</Text>
          </TouchableOpacity>
          <TouchableOpacity
            // onPress={}
            style={styles.buttonContainer}
            activeOpacity={0.7}
          >
            <Text style={styles.buttonText}>Acessar Visitas</Text>
          </TouchableOpacity>
        </View>
      </SideMenu>
    );
  }
}

const { shape, func, number } = React.PropTypes;

SchoolInfoScreen.propTypes = {
  setSchoolInfo: func.isRequired,
  school: shape({
    schoolCode: number.isRequired,
  }).isRequired,
};

export default SchoolInfoScreen;
