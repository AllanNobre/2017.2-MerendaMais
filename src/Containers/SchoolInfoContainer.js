import { connect } from 'react-redux';
import SchoolInfoScreen from '../screens/SchoolInfoScreen';
import { setSchoolInfo } from '../actions/schoolActions';

const mapStateToProps = state => (
  {
    school: state.school,
    isLoading: state.application.isLoading,
    message_erro: state.application.message_erro,
  }
);

const mapDispatchToProps = dispatch => ({
  setSchoolInfo: schoolData => dispatch(setSchoolInfo(schoolData)),
});


const SchoolInfoContainer = connect(mapStateToProps, mapDispatchToProps)(SchoolInfoScreen);

export default SchoolInfoContainer;
