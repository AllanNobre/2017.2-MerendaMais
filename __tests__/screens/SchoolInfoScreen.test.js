import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import SchoolInfoContainer from '../../src/Containers/SchoolInfoContainer';
import SchoolInfoScreen from '../../src/screens/SchoolInfoScreen';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  application: {
    isLoading: false,
    message_erro: '',
  },
  school: {
    schoolCode: 123,
    schoolName: 'Teste',
    schoolPhone: '11111111111',
    schoolEmail: 'teste@teste.com',
    schoolLat: '',
    schoolLong: '',
    schoolStudents: '1000',
  },
  counselor: {
    profile: {
      CAE: 'Distrito Federal',
    },
  },
};

const store = mockStore(initialState);

describe('Testing SchoolInfoScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <SchoolInfoContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Testing SchoolInfo buttons', () => {
  it('Test if goToMaps Button is rendered', () => {
    const wrapper = shallow(<SchoolInfoScreen {...initialState} />);
    const button = wrapper.findWhere(c => c.key() === 'renderButton');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });
});
