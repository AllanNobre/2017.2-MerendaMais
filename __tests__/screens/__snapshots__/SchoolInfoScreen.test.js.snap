// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SchoolInfoScreen renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF9500",
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        color="black"
        name="ios-arrow-back-outline"
        size={45}
        style={
          Object {
            "margin": 8,
          }
        }
      />
    </TouchableOpacity>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "flex": 1,
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Pesquisar Escola
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="navicon"
        size={32}
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 20,
            "paddingLeft": 2,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="black"
          name="location-city"
          size={32}
          style={
            Object {
              "margin": 8,
            }
          }
        />
        <TextInput
          keyboardType="default"
          maxLength={50}
          onChangeText={[Function]}
          placeholder="Estado/MunicÃ­pio"
          returnKeyType="go"
          underlineColorAndroid="transparent"
          value=""
          width={280}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 20,
            "paddingLeft": 2,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="black"
          name="search"
          size={30}
          style={
            Object {
              "margin": 8,
            }
          }
        />
        <TextInput
          keyboardType="default"
          maxLength={50}
          onChangeText={[Function]}
          placeholder="Escola a pesquisar"
          returnKeyType="go"
          underlineColorAndroid="transparent"
          value=""
          width={280}
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "gray",
            "borderRadius": 7,
            "borderWidth": 1,
            "marginTop": 15,
            "paddingLeft": 2,
          }
        }
      >
        <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue=""
        >
          <PickerItem
            color="#95a5a6"
            label="Escolha sua UF"
            value=""
          />
          <PickerItem
            label="DF"
            value="DF"
          />
        </Picker>
      </View>
    </View>
    <View
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 7,
          "borderWidth": 1,
          "flex": 2.5,
          "justifyContent": "center",
          "width": 320,
        }
      }
    >
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>
    <View
      key="renderButton"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 20,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        disabled={true}
        style={
          Object {
            "backgroundColor": "#DEDEDE",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginBottom": 0,
            "marginTop": 50,
            "paddingHorizontal": 117,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Pesquisar
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
